api/envoy/extensions/filters/http/geoip/v3/geoip.proto

```diff
@@ -21,64 +21,74 @@ option (xds.annotations.v3.file_status).work_in_progress = true;
+    string region = 3
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+
     // If set, the header will be used to populate the ASN associated with the IP address.
-    string asn = 4 [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+    string asn = 4
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+
     // If set, the IP address will be checked if it belongs to any type of anonymization network (e.g. VPN, public proxy etc)
     // and header will be populated with the check result.
-    string is_anon = 5 [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+    string is_anon = 5
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+
     // If set, the IP address will be checked if it belongs to a VPN and header will be populated with the check result.
-    string anon_vpn = 6 [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+    string anon_vpn = 6
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+
     // If set, the IP address will be checked if it belongs to a hosting provider and header will be populated with the check result.
-    string anon_hosting = 7 [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+    string anon_hosting = 7
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+
     // If set, the IP address will be checked if it belongs to a TOR exit node and header will be populated with the check result.
-    string anon_tor = 8 [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+    string anon_tor = 8
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+
     // If set, the IP address will be checked if it belongs to a public proxy and header will be populated with the check result.
-    string anon_proxy = 9 [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
+    string anon_proxy = 9
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
   }
 
   message GeolocationProvider {
     // The name of the geoip driver to instantiate. The name must match a
-    // supported geoip driver. Built-in drivers:
-    //
-    // - *envoy.geoip_providers.maxmind*
-    string name = 1 [(validate.rules).string = {min_bytes: 1}];
+    // supported geoip driver.
+    string name = 1 [(validate.rules).string = {min_len: 1}];
 
     // Geoip driver specific configuration which depends on the driver being instantiated.
-    // See the geoip drivers for examples:
-    //
-    // - :ref:`MaxMindConfig <envoy_api_msg_config.geoip.v3.MaxMindConfig>`
-    // [#extension-category: envoy.geoip_providers]
     oneof config_type {
```

Comment:
Why is this in a oneof? An Any type is kind of an implicit version of the same thing.